libs/javalib/src/mill/javalib/MavenPublishModule.scala

```diff
@@ -0,0 +1,75 @@
+package mill.javalib
+
+import com.lihaoyi.unroll
+import mill.*
+import mill.api.*
+import mill.javalib.PublishModule.PublishData
+import mill.util.Tasks
+
+trait MavenPublishModule extends PublishModule, MavenWorkerSupport, SonatypeCredentialsModule,
```

Comment:
If `SonatypeCredentialsModule` is being used for non-sonatype publishing as well, we should mention that in its scaladoc

Reply:
I actually had not considered this inconsistency. If we continue with my current implementation, we would actually expect the user to provide the environment variables `MILL_SONATYPE_USERNAME` and `MILL_SONATYPE_PASSWORD` for publishing to "plain" Maven as well. I guess I was subconsciously thinking "it will also be Sonatype". I am not sure what the right path forward is here. Should we keep it as is and just add the scaladoc? Or should we maybe rename this module and make the environment variable different for Sonatype and "plain" Maven?